### PR TITLE
Make big, flexable and readable ESLint config

### PR DESCRIPTION
**Pull request by PoorTable**
This PR shows idea mentioned in https://github.com/brains-and-beards/react-native-template-redbeard/issues/27

### CODE REVIEW

File: template/.eslintrc.js [hunk 1]
```diff
@@ -6,30 +6,238 @@ module.exports = {
...
-    'react/prop-types': 'off',
     'sort-imports': 'off',
+    // General
+    'prettier/prettier': 'error',
+    'import/order': 'off',
+    'no-console': ['error', { allow: ['warn', 'error'] }],
+    'no-cond-assign': 1, // disallow assignment in conditional expressions
+    'no-const-assign': 2, // disallow assignment to const-declared variables
+    'no-constant-condition': 0, // disallow use of constant expressions in conditions
+    'no-control-regex': 1, // disallow control characters in regular expressions
```
Comment (sharnik) [inline template/.eslintrc.js]: I often use `\A` and `\z`. Why would we not allow control characters?

File: template/.eslintrc.js [hunk 2]
```diff
@@ -6,30 +6,238 @@ module.exports = {
...
+    // The following rules are made available via `eslint-plugin-react-native`
+
+    'react-native/no-inline-styles': 1,
+
+    // Jest Plugin
+    // The following rules are made available via `eslint-plugin-jest`.
+    'jest/no-disabled-tests': 1,
+    'jest/no-focused-tests': 1,
+    'jest/no-identical-title': 1,
+    'jest/valid-expect': 1,
```
Comment (sharnik) [inline template/.eslintrc.js]: Are all those rules that you have a strong opinion that we should change? 🤔

**Review by sharnik — COMMENTED** [review]
It seems like you're proposing to change 200 eslint rules at once 🤔

It would take a lot of time to get the whole team's alignment on it. What are we really trying to achieve here?

File: template/.eslintrc.js (hunk 2 above)
Comment (PoorTable) [reply template/.eslintrc.js]: It's copied rules, as I remember)

File: template/.eslintrc.js [hunk 3]
```diff
@@ -6,30 +6,68 @@ module.exports = {
...
+    'prettier/prettier': 'error',
     'import/order': 'off',
     'no-console': ['error', { allow: ['warn', 'error'] }],
+
+    // ESLint Comments Plugin
+    // The following rules are made available via `eslint-plugin-eslint-comments`
+    'eslint-comments/no-aggregating-enable': 1, // disallows eslint-enable comments for multiple eslint-disable comments
+    'eslint-comments/no-unlimited-disable': 1, // disallows eslint-disable comments without rule names
+    'eslint-comments/no-unused-disable': 1, // disallow disables that don't cover any errors
+    'eslint-comments/no-unused-enable': 1, // // disallow enables that don't enable anything or enable rules that weren't disabled
```
Comment (sharnik) [inline template/.eslintrc.js:54]: Can we simplify that using 
```
 "extends": [
// …
        "plugin:eslint-comments/recommended"
    ],
```
instead? Or do you want to tweak those rules?

File: template/.eslintrc.js [hunk 4]
```diff
@@ -6,30 +6,68 @@ module.exports = {
...
     jest: true,
     node: true,
   },
+  globals: {
+    React: true,
+    JSX: true,
+  },
   rules: {
+    // Typescript Plugin
+    // The following rules are made available via `@typescript-eslint`.
```
Comment (sharnik) [inline template/.eslintrc.js]: > The following rules are made available via `X`.

I don't think those comments (here and below for other plugins) are necessary. All those rules start with the plugin name anyway (`'@typescript-eslint/blah-blah'`).

File: template/.eslintrc.js [hunk 5]
```diff
@@ -6,30 +6,68 @@ module.exports = {
...
-  plugins: ['@typescript-eslint', 'babel', 'react-hooks', 'react-native', 'react', 'prettier'],
+  plugins: [
+    '@typescript-eslint',
+    'babel',
+    'react-hooks',
+    'react-native',
+    'react',
+    'prettier',
+    'jest',
+    'eslint-comments',
```
Comment (sharnik) [inline template/.eslintrc.js:17]: Good additions 👍

**Review by sharnik — APPROVED** [review]

File: template/.eslintrc.js (hunk 3 above)
Comment (PoorTable) [reply template/.eslintrc.js:54]: For some reason (probably `coz of eslint recommended override) with this variant rules are not working (not reporting the warning/errors)

**Pull request merged**